velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -639,4 +639,38 @@ struct ArrayFlattenFunction {
     }
   }
 };
+
+/// This class implements the array remove function.
+///
+/// DEFINITION:
+/// array_remove(x, element) → array
+/// Remove all elements that equal element from array x.
+template <typename TExecCtx, typename T>
+struct ArrayRemoveFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExecCtx);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Array<T>>& out,
+      const arg_type<Array<T>>& array,
+      const arg_type<T>& element) {
+    int64_t elementCount = 0;
+    for (const auto& item : array) {
+      if (element != item.value()) {
+        elementCount++;
+      }
+    }
+    out.reserve(elementCount);
+
+    for (const auto& item : array) {
+      if (element != item.value()) {
+        if (item.has_value()) {
+          auto& newItem = out.add_item();
```

Comment:
FYI: for primitives you can directly do out.push_back(item.value()) and for nulls you can do out.push_back(std::nullopt).

it would be nice to allow out.add_item(item); to be handled where item is optional accessor but we do have that 
now,. will file task for it. that will reduce 6 lines here to one.

Reply:
Has this comment been addressed?

Reply:
Done, Currently I used `call(Out& out, const In& inputArray,...`, so still use  `out.add_item()`(in this case, out.push_back() doesn`t work)

Reply:
out.push_back(item.value()) did not work?